gmime/meson.build

```diff
@@ -0,0 +1,221 @@
+  'gmime-encodings.h',
+  'gmime-error.h',
+  'gmime-filter-basic.h',
+  'gmime-filter-best.h',
+  'gmime-filter-charset.h',
+  'gmime-filter-checksum.h',
+  'gmime-filter-dos2unix.h',
+  'gmime-filter-enriched.h',
+  'gmime-filter-from.h',
+  'gmime-filter-gzip.h',
+  'gmime-filter.h',
+  'gmime-filter-html.h',
+  'gmime-filter-openpgp.h',
+  'gmime-filter-smtp-data.h',
+  'gmime-filter-strip.h',
+  'gmime-filter-unix2dos.h',
+  'gmime-filter-windows.h',
+  'gmime-filter-yenc.h',
+  'gmime-format-options.h',
+  'gmime-gpg-context.h',
+  'gmime.h',
+  'gmime-header.h',
+  'gmime-iconv.h',
+  'gmime-iconv-utils.h',
+  'gmime-message.h',
+  'gmime-message-part.h',
+  'gmime-message-partial.h',
+  'gmime-multipart-encrypted.h',
+  'gmime-multipart.h',
+  'gmime-multipart-signed.h',
+  'gmime-object.h',
+  'gmime-param.h',
+  'gmime-parser.h',
+  'gmime-parser-options.h',
+  'gmime-part.h',
+  'gmime-part-iter.h',
+  'gmime-pkcs7-context.h',
+  'gmime-references.h',
+  'gmime-signature.h',
+  'gmime-stream-buffer.h',
+  'gmime-stream-cat.h',
+  'gmime-stream-file.h',
+  'gmime-stream-filter.h',
+  'gmime-stream-fs.h',
+  'gmime-stream-gio.h',
+  'gmime-stream.h',
+  'gmime-stream-mem.h',
+  'gmime-stream-mmap.h',
+  'gmime-stream-null.h',
+  'gmime-stream-pipe.h',
+  'gmime-text-part.h',
+  'gmime-utils.h',
+  'gmime-version.h',
+  'internet-address.h',
+]
+
+inc_conf = include_directories('..', '../util')
+
+version_data = configuration_data()
+version_data.set('GMIME_MAJOR_VERSION', GMIME_MAJOR_VERSION)
+version_data.set('GMIME_MINOR_VERSION', GMIME_MINOR_VERSION)
+version_data.set('GMIME_MICRO_VERSION', GMIME_MICRO_VERSION)
+version_data.set('GMIME_BINARY_AGE', GMIME_BINARY_AGE)
+version_data.set('GMIME_INTERFACE_AGE', GMIME_INTERFACE_AGE)
+configure_file(input: 'gmime-version_meson.h.in',
+  output: 'gmime-version.h',
+  configuration : version_data
+)
+
+gen_table = executable('gen-table', 'gen-table.c')
+charset_map = executable('charset-map', 'charset-map.c', dependencies : deps, 
+  include_directories: inc_conf)
+
+# For the future, we can generate the source code at runtime instead of having
+# it in the filesystem
+# table_h = custom_target('gen-table',
+#   capture: true,
+#   output : ['gmime-table-private.h'],
+#   command : [gen_table])
+# charset_h = custom_target('charset-map',
+#   capture: true,
+#   output : ['gmime-charset-map-private.h'],
+#   command : [charset_map])
+#
+# libgmime = library(gmime_package, libgmime_src + table_h + charset_h, dependencies : deps, 
+#   include_directories: inc_conf, link_with: libutil, install: true)
+
+libgmime = library(gmime_package, libgmime_src, dependencies : deps, 
+  include_directories: inc_conf, link_with: libutil, install: true)
+
+pkg = import('pkgconfig')
+
+pkg.generate(libgmime,
+  name: meson.project_name(),
+  filebase: gmime_package,
+  subdirs: gmime_package,
+  description: 'MIME parser and utility library',
+  requires: LIBS,
+  # libraries_private: EXTRA_LIBS,
```

Comment:
LIBS is a list of strings. Consider using the returned `dependency()` objects as those automatically do "the right thing" (including enforce the same version constraints you used when looking up the dependency to begin with).

Reply:
Fixed